Automattic/jetpack
# Blocks: load different bundle in the block-based widget editor

Related: #20357

#### Changes proposed in this Pull Request:

As described in #20357, we cannot use blocks that rely on post information and the `@wordpress/editor` package in the block-based widget editor that will ship in WordPress 5.8.

We happen to have an existing alternative bundle, that was built for P2, and that had a similar goal: see #15467 and D41999-code for more info. One potential solution to #20357 would be to use that same bundle in the widget editor.
This PR leverages that existing bundle, and updates it to remove the blocks that are problematic in the widget editor.

This has an advantage: we do not have to come up with a new solution.
And some disadvantages:

- Although named `no-post-editor`, I believe that bundle mostly meant "No blocks that only appear in the sidebar". I believe we may be able, and want to be using blocks such as our Video block or the Tiled Gallery block on P2. cc @lezama.
- Additionally, it also means that from now on, we would have to manually maintain the 2 bundles and remove where each one belongs. That same problem was highlighted in https://github.com/Automattic/jetpack/pull/15467#issuecomment-614740859
- It's also a problem for folks using the `jetpack_blocks_variation` filter to overwrite the block variation used anywhere on a site. Maybe we should force the use of the no-post-editor bundle after that filter has been applied.

#### Jetpack product discussion

* p1626278473009100-slack-CBTN58FTJ

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Start from a site using WP 5.8 RC
* Go to Appearance > Widgets
* You should see some Jetpack blocks available there.
* The Tiled gallery block should not be available.
* You should see no errors in your logs.
